Testing: remove policy restriction for `test_default_hash` #7334

0xquark opened the pull request:
The test_default_hash was being skipped unless the policy was ATLAS, but ATLAS does not have a specific lfn2pfn algorithm. The default hash algorithm should be tested universally.

Removed the @pytest.mark.skipif condition to ensure the test runs in all conditions.

Fixes : #7334 

<!-- Please read https://rucio.cern.ch/documentation/contributing before submitting a pull request -->

0xquark committed:
Testing: remove policy restriction for test_default_hash; Fix #7334

The test_default_hash was being skipped unless the policy was ATLAS,
but ATLAS does not have a specific lfn2pfn algorithm. The default
hash algorithm should be tested universally.

Removed the @pytest.mark.skipif condition to ensure the test runs
in all conditions.

Signed-off-by: Karanjot Singh <drquark@duck.com>

0xquark commented:
The test is failing for Belle2 but passing for ATLAS. Can someone share the Belle2 policy package with me? I want to check the hashing algorithm used in that policy. 

I found a temporary Belle2 policy package ([link](https://github.com/rucio/temporary-belle2-policy-package/blob/main/src/belleii_rucio_policy_package/lfn2pfn.py)), but it has its own "deterministic" translation that isn’t truly deterministic. It inherits from `RSEDeterministicTranslation` but overrides its behavior, returning the `name` instead. That’s why, in the test, it returns `bar` ([link](https://github.com/rucio/rucio/actions/runs/13132682037/job/36653775093?pr=7381)), which is just what was passed in.  

Can someone confirm if this is the cause of the failure? Should we update the Belle2 package to use the default hashing algorithm, or should I modify the test to account for Belle2 returning `bar`?

bari12 commented:
This test doesn't make much sense for Belle II, since they do not use the hash lfn2pfn. All the BelleII `name`s are already full pathes (`/a/b/c/d/file.txt`) so it doesn't make sense to add an extra transformation for it.

0xquark committed:
[testing] Skip default hash test for BelleII policy

Belle II does not use the hashed lfn2pfn algorithm, as all file names are already full paths.
This test is only relevant for policies that apply hashing (e.g., ATLAS).

- Added a skip condition for BelleII in test_default_hash.

Signed-off-by: Karanjot Singh <drquark@duck.com>

0xquark commented:
Thanks @bari12! I’ve skipped this test for BelleII since it doesn’t use the hashed lfn2pfn.
